cvat-ui/src/components/quality-control/quality-control-page.tsx

```diff
@@ -166,89 +208,133 @@ function QualityControlPage(): JSX.Element {
-                settings.iouThreshold = values.iouThreshold / 100;
-                settings.compareAttributes = values.compareAttributes;
-                settings.emptyIsAnnotated = values.emptyIsAnnotated;
-
-                settings.oksSigma = values.oksSigma / 100;
-                settings.pointSizeBase = values.pointSizeBase;
-
-                settings.lineThickness = values.lineThickness / 100;
-                settings.lineOrientationThreshold = values.lineOrientationThreshold / 100;
-                settings.orientedLines = values.orientedLines;
-
-                settings.compareGroups = values.compareGroups;
-                settings.groupMatchThreshold = values.groupMatchThreshold / 100;
-
-                settings.checkCoveredAnnotations = values.checkCoveredAnnotations;
-                settings.objectVisibilityThreshold = values.objectVisibilityThreshold / 100;
-
-                settings.panopticComparison = values.panopticComparison;
-                try {
-                    dispatch(reducerActions.setQualitySettingsFetching(true));
-                    const responseSettings = await settings.save();
-                    dispatch(reducerActions.setQualitySettings(responseSettings));
-                    notification.info({ message: 'Settings have been updated' });
-                } catch (error: unknown) {
-                    notification.error({
-                        message: 'Could not save quality settings',
-                        description: typeof Error === 'object' ? (error as object).toString() : '',
-                    });
-                    throw error;
-                } finally {
-                    dispatch(reducerActions.setQualitySettingsFetching(false));
-                }
+            dispatch(reducerActions.setQualitySettingsFetching(true));
+            const updatedSettings = await Promise.all(settingsList.map((setting) => setting.save()));
+            const updatedInstanceSettings = updatedSettings.find((setting) => setting.id === settings.id) || settings;
+            const updatedChildrenSettings = childrenSettings?.map((childSetting) => {
+                const updatedSetting = updatedSettings.find((setting) => setting.id === childSetting.id);
+                return updatedSetting || childSetting;
+            }) || null;
+
+            dispatch(reducerActions.setQualitySettings(updatedInstanceSettings, updatedChildrenSettings));
+            notification.info({ message: 'Settings have been updated' });
+        } catch (error: unknown) {
+            notification.error({
+                message: 'Could not save quality settings',
+                description: typeof Error === 'object' ? (error as object).toString() : '',
+            });
+            if (onError) {
+                onError();
             }
-            return settings;
-        } catch (e) {
-            return false;
+            throw error;
```

Comment:
Why do we need both `onError` and re-throw?
3rdparty code calling this function may .catch() error or pass `onError` callback. Just need to choose something one.

Reply:
Removed callback, only re-throw left